### ISSUE TITLE
Browser does not display changes (e.g. in HTML files)

### ISSUE THREAD

**Issue by whatis777** [jhipster/generator-jhipster#5096]
##### **Overview of the issue**

Browser does not display changes in HTML files. Only after 'yo jhipster' the changes are shown in the browser.
It's not a browser cache issue as I checked that and used different browsers

##### **Motivation for or Use Case**

As a developer I want to see changes in HTML files without regenerating the application with 'yo jhipster'

##### **Reproduce the error**

1. Create an app with jHipster (see configuration below)
2. Start the app: ./mvnw
3. Change HTML file (e.g. app title)
4. Reload page in browser
=> no change visible
5. Run 'yo jhipster'
=> change visible

##### **Related issues**

could not find related issues, sorry

##### **Suggest a Fix**

n.a.

##### **JHipster Version(s)**

4.0.1

##### **JHipster configuration**

gedcommon@0.0.0 /home/novotny/development/projects/gedcommon
├── UNMET PEER DEPENDENCY @angular/compiler@2.4.4
├── UNMET PEER DEPENDENCY @angular/core@2.4.4
└── generator-jhipster@4.0.0 

yaml
{
  "generator-jhipster": {
    "jhipsterVersion": "4.0.0",
    "baseName": "gedcommon",
    "packageName": "de.ibn.gedcommon",
    "packageFolder": "de/ibn/gedcommon",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "no",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mariadb",
    "searchEngine": false,
    "messageBroker": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "rememberMeKey": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": false,
    "serviceDiscoveryType": false,
    "clientFramework": "angular2"
  }
}

openjdk version "1.8.0_121"
OpenJDK Runtime Environment (build 1.8.0_121-8u121-b13-0ubuntu1.16.04.2-b13)
OpenJDK 64-Bit Server VM (build 25.121-b13, mixed mode)
git version 2.7.4
node: v7.5.0
npm: 4.1.2
bower: 1.8.0
gulp:
[21:12:50] CLI version 3.9.1
[21:12:50] Local version 3.9.1
yeoman: 1.8.5
yarn: 0.19.1

##### **Entity configuration(s)**

none

##### **Browsers and Operating System**

Ubuntu 16.04 LTS, 64bit
Firefox & Chrome

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by jdubois** [comment]
Don't take this harshly, but you're like the 10th person to tell this to us.
You need to run `yarn start` to have hot reload enabled, this is written everywhere, but it looks like it's not clear for many people - and I don't know what to do here!! I'll try to make it more clear in the documentation.

**Comment by whatis777** [comment]
If this happens to more people then maybe there's really a lack of documentation and you could improve it? You say you don't know what to do there. Well, in "Creating an application" chapter it would have been very useful, if there were a hint to `yarn start` - but only 'mvnw' is mentioned.

**Comment by baskeboler** [comment]
@whatis777 check out https://jhipster.github.io/development/

**Comment by jdubois** [comment]
It's also printed out in the console when you generate, and written in your generated "README" file... But yes we need to write it even more.

**Comment by pascalgrimaud** [comment]
@whatis777 :

When using Angular2, people have to understand:
- if they modify files in src/main/java folder, they have to recompile the backend -> you can use spring devtools for hot reload
- if they modify file in src/main/webapp folder, they have to recompile the frontend -> or you can use `yarn start` as mentionned everywhere

**Comment by jdubois** [comment]
I have put more warnings on the website, see https://github.com/jhipster/jhipster.github.io/commit/4a1415fde33db9e04224d8d25146e544fa0ddbee

**Comment by jdubois** [comment]
Another important solution would be a new video, @mraible said he would be able to make one soon (and otherwise I will do it at some point)

**Comment by whatis777** [comment]
@jdubois and @pascalgrimaud: Thanks for the information and for adding this to the documentation.

**Comment by jdubois** [comment]
Thanks for reporting this @whatis777 - I know it's not trivial for end-users... We've been working on this for one year, so what is "normal" to us isn't obvious for others...

**Comment by mraible** [comment]
@jdubois I was going to record a screencast today, but I can't deploy to Heroku with 4.0.1.

```
Creating Heroku deployment files
events.js:160
      throw er; // Unhandled 'error' event
      ^

ReferenceError: /Users/mraible/blog/node_modules/generator-jhipster/generators/heroku/templates/_Procfile:1
 >> 1| web: java $JAVA_OPTS <% if (applicationType == 'gateway' || dynoSize == 'Free') { %>-Xmx256m<% } %> -jar <% if (buildTool == 'maven') { %>target<% } %><% if (buildTool == 'gradle') { %>build/libs<% } %>/*.war --spring.profiles.active=prod,heroku --server.port=$PORT <% if (prodDatabaseType == 'mongodb') { %>--spring.data.mongodb.database=$(echo "$MONGODB_URI" | sed "s/^.*:[0-9]*\///g")<% } %>
    2|

dynoSize is not defined
    at eval (eval at compile (/Users/mraible/blog/node_modules/ejs/lib/ejs.js:495:12), <anonymous>:10:44)
    at returnedFn (/Users/mraible/blog/node_modules/ejs/lib/ejs.js:524:17)
    at Object.exports.render (/Users/mraible/blog/node_modules/ejs/lib/ejs.js:332:37)
    at process (/Users/mraible/blog/node_modules/mem-fs-editor/lib/actions/copy-tpl.js:12:18)
    at applyProcessingFunc (/Users/mraible/blog/node_modules/mem-fs-editor/lib/actions/copy.js:13:16)
    at EditionInterface.exports._copySingle (/Users/mraible/blog/node_modules/mem-fs-editor/lib/actions/copy.js:51:16)
    at EditionInterface.exports.copy (/Users/mraible/blog/node_modules/mem-fs-editor/lib/actions/copy.js:23:17)
    at EditionInterface.module.exports [as copyTpl] (/Users/mraible/blog/node_modules/mem-fs-editor/lib/actions/copy-tpl.js:10:8)
```

**Comment by pascalgrimaud** [comment]
@mraible : it should be fixed by https://github.com/jhipster/generator-jhipster/pull/5106

**Comment by jdubois** [comment]
Yes I can release a 4.0.2 version tomorrow if you need it

Le 5 févr. 2017 7:21 PM, "Pascal Grimaud" <notifications@github.com> a
écrit :

**Comment by ruddell** [comment]
@thinkocapo If you have an issue, please open a new one and fill out the template.  This was closed over a year ago so it's probably not related.

------

Please don't comment on an **old issue**.
If you think this issue still applies, please [create a new ticket](https://github.com/jhipster/generator-jhipster/issues/new) with proper details.

**If you have a question** please use Stack Overflow, and tag the question with [jhipster](http://stackoverflow.com/questions/tagged/jhipster). This helps the project to keep the issue tracker clean. Also, Stack Overflow will give your question a larger audience:

- This will increase your chances to get an answer
- Answers will be of higher quality, as there is a voting system
- This will also help other users who might have the same issue, as questions are tagged and easily searchable
